### DISCUSSION TITLE
Ill posed problem

### DISCUSSION THREAD

**Discussion by Moradielham** [yalmip/YALMIP#1164] [General discussion]
Dear professor, I have a state feedback code that works well. But when applying changes 
Anew=(Aold 0 ;C 0)  Bnew=(Bold;0) and ...to turn it into a PI-state feedback controller, I encounter an ill posed problem. How can I solve this problem?

**Comment by johanlofberg** [comment]
Impossible to answer. Balanced realization before any computations is a generic trick though

**Reply by Moradielham** [reply]
Dear professor, I think I should have explained more. I have already written a code for state feedback in yalmip and I have tested it and it works well (I asked you about it before). According to the relationships and equations in the articles, we want to convert the state feedback controller to PI like state feedback. In this way, there are changes in the state matrices that I sent to the attachment. By applying these changes, the solution status becomes ill posed and gamma becomes around 0.15. My question is, why did it happen and how can it be resolved?  Thank you for your reply

**Reply by Moradielham** [reply]
![Screenshot_۲۰۲۲-۰۸-۱۲-۱۵-۵۳-۳۵-۸۸۳_com google android apps docs](https://user-images.githubusercontent.com/106932603/184345301-39e5f72e-c6b7-4d69-8ad6-1a2be3badc7a.jpg)

**Reply by Moradielham** [reply]
In addition,  X and Y (lmi variables) are very small. And some eigenvalues of X ​​are also zero.

**Reply by johanlofberg** [reply]
We have no idea what optimization problem you are solving. If variables are close to 0, and you don't think that is reasonable, it means you are missing some constraints. it could also be that you have posed an infeasible problem, such as replacing X>0 with X>=e*I for some e>0, when in reality only psd solutions are possible, and the solver just gives up. There are a million different things that can go wrong, all the way from errors in the paper, you doing something wrong, to solver just having issues.

**Reply by Moradielham** [reply]
This is my code, I would be grateful if you could take a look at it. When I added the section"%converting to pi controller" to it and subsequently resized the other matrices, The answer was ill posed

**Reply by johanlofberg** [reply]
Code must be copied into codeblocks <> otherwise it gets broken

**Reply by Moradielham** [reply]
Does it work?

**Reply by johanlofberg** [reply]
Just try to copy it to matlab, it is broken

y = 5*x
z = 7*y

vs

```
y = 5*x
z = 7*y
```

**Reply by Moradielham** [reply]
I selected <> and copied the code into '', is there anything else I should do?

**Reply by johanlofberg** [reply]
copy code from matlab, paste, select it and press <>

**Reply by johanlofberg** [reply]
Already this fails

`optimize(Fi <= 0)`

slacking it with

```
sdpvar t
optimize(Fi <= t*eye(188),t)
```

shows that minimal t is significantly larger than zero. Either due to some serious error in your theory, or a consequence of bad data (huge coefficients)

```
>> Fi
Linear matrix variable 188x188 (symmetric, real, 982 variables)
Coeffiecient range: 0.0052429 to 2397848850.2471
```

**Reply by Moradielham** [reply]
I think it's because of bad data. It was working well before making the changes and converting it to a PI-controller. But by applying the changes related to the pi controller, a part was added to the previous matrices, which has much smaller data compared to the previous matrix, this happened. How can such a problem caused by bad data be solved?

**Reply by johanlofberg** [reply]
That's for you to figure out. No standard fix.

**Reply by Moradielham** [reply]
Thank you for your useful help

**Reply by Moradielham** [reply]
Dear professor, excuse me. For what reasons may the solution status be unknown?

**Reply by johanlofberg** [reply]
Numerically problematic, such has bad coefficients in the data, or very small feasible set with no interior, or just generally hard ill-posed/ill-conditioned

**Reply by Moradielham** [reply]
Thank you very much

**Reply by Moradielham** [reply]
Excuse me again, if we want to define a n*n matrix that is not symmetric and is a so-called general matrix, how do we do it?

**Reply by johanlofberg** [reply]
First section on variables
https://yalmip.github.io/tutorial/basics/
